app/models/puppet_report.rb

```diff
@@ -0,0 +1,20 @@
+class PuppetReport < Report
+  def self.model_name
+    superclass.model_name
```

Comment:
why is that needed? IMHO you can use include Foreman::STI to get rid of most related issues?